Release notes 20220221 (#336)
circonus/docs: content/circonus/on-premises/changelog/2022.md

* No user-facing changes.

### data_storage

Related roles: caql_broker, fault_detection, stratcon

Version: 0.1.1645029602 (6896be6)

**Function Removal Notice**

> Metric Search v2 and Metric Clusters have been deprecated for some time.  We
> plan to remove these deprecated functions soon for the SaaS site, and in the
> subsequent Inside release. This will affect CAQL checks as well as CAQL
> Datapoints on graphs. The UI will now show users a warning when one of the
> deprecated functions is used. Circonus offers a more powerful tag-search
> feature, exposed as `find()` in CAQL.

* Graphite find endpoint will no longer return a 500 on a bad request, favoring
  the more accurate 400 error.
* Fix crash when attempting to fetch histogram shard data from the
  `/histogram_raw` endpoint when histogram shards are not enabled.
* Fix crash when enabling the monitor module and using the legacy histogram
  database.
* Gossip messages modified to be larger and fewer in number.
* DELETE on `/full/tags` now accepts `X-Snowth-Advisory-Limit` properly to
  limit the number of deletes that will occur in one operation.
  * There is also a config-based maximum that this value may be, and beyond

>     which the limit set by `X-Snowth-Advisory-Limit` will not be respected.

Reviewer: I think the default here is now 10k

  * Additionally, this endpoint now returns the header
    `X-Snowth-Incomplete-Results: "true"` if the requested delete operation
    requires more deletes than allowed by `X-Snowth-Advisory-Limit` or the
    server-configured maximum. This helps ensure that very large deletes will

>     not consume memory ad infinitum.

Reviewer: ```suggestion
    not consume memory ad infinitum.
    NOTE:   This is a change in behavior.  Endusers utilizing full delete in their scripts, will want to put
    this call into a loop to maintain existing expected behavior.   e.g.
    do {
       resposne = call-delete-api()
    } while( response->header->'X-Snowth-Incomplete-Results: "true"' )
```

* Add explicit `histogram:random:` functions for each supported CDF.
* Fix bug where jlog subscribers could hang around for too long, causing
  journal data that should have been removed to remain on disk until the next
  time the node restarts.

### fault_detection

Version: 0.1.1639089770 (90cc626)

* No changes since [2022-01-10](/circonus/on-premises/changelog/2022#2022-01-10)

### FQ

Related roles: caql_broker, fault_detection, mq, stratcon, web_stream

Version: 0.13.10

* No changes since [2021-05-03](/circonus/on-premises/changelog/2021#2021-05-03)

### GoAPI

Related roles: api

Version: 0.10.23

* Add re-connection support for IRONdb proxy dial attempts.
* Add support for metric search match impl options.
* Correct an error causing metric search graphs not to save correctly.
* Add the `_check_module` and `_check_target` fields to the response data
  returned by the `/alert` endpoint.
* Correct an error in the `/graph` endpoint handling of stacks.
* Improve error messaging when proxy requires an account ID.

> * Add the new broker check module type support to the `/check_bundle` endpoint.

Reviewer: ```suggestion
* Add support for future broker check module type support to the `/check_bundle` endpoint.
```

### Hooper

Version: 0.1.1643896828 (a06c0b8)

* No changes since [2022-02-07](/circonus/on-premises/changelog/2022#2022-02-07)

### libmtev

Related roles: broker, caql_broker, data_storage, fault_detection, stratcon, web_stream

Version: 2.2.0

* Deprecate `mtev_atomic` functionality.
* Added `MTEV_MAYBE_REALLOC_WITH_TYPE` for C++ compatibility. The older version
  is maintained for backwards compatibility.

### notification

Version: 0.1.1644263161 (1e0f82e)

* Create and send alerts when Postgres database is unavailable, store them when
  Postgres is back.
* Configurable log size and count via `resources.conf`.

### Reconnoiter

Related roles: broker, caql_broker, data_storage, fault_detection, stratcon, web_stream

Version: 0.1.1644523407 (c08cd36)

* (libnoit) Tag searches can be (much) longer than tags.
  * A Graphite-style tag search could have thousands of `{opt1,opt2,opt3}`
    expansion internals making the search expression very long while preserving

>     a reasonable final matching tag cat or name. We should allow the AST to

Reviewer: ```suggestion
    a reasonable final matching tag cat or name.  Allow the AST to
```
